packages/network-controller/tests/provider-api-tests/no-block-param.ts

```diff
@@ -0,0 +1,916 @@
+/* eslint-disable jest/require-top-level-describe, jest/no-export, jest/no-identical-title, jest/no-if, jest/expect-expect */
+
+import {
+  MockCommunications,
+  ProviderType,
+  waitForNextBlockTracker,
+  waitForPromiseToBeFulfilledAfterRunningAllTimers,
+  withMockedCommunications,
+  withNetworkClient,
+} from './helpers';
+import {
+  buildFetchFailedErrorMessage,
+  buildInfuraClientRetriesExhaustedErrorMessage,
+} from './shared-tests';
+
+const returnsErrorAfterRetries = async (
```

Comment:
Future PR: None of the other tests have a custom expectation helper, and we only call this function three times in this file, so I wonder if it makes sense to extract it. I'm going to keep this the same in the extension for now.